src/libvirt-dbus.js

```diff
@@ -1450,6 +1334,56 @@ export function changeNetworkAutostart(network, autostart, dispatch) {
             .then(() => dispatch(getNetwork({ connectionName: network.connectionName, id: network.id, name: network.name })));
 }
 
+export function changeNetworkSettings({
+    name,
+    id: objPath,
+    connectionName,
+    hotplug,
+    persistent,
+    macAddress,
+    newMacAddress,
+    networkType,
+    networkSource,
+    networkModel,
+    dispatch,
+}) {
+    /*
+     * 0 -> VIR_DOMAIN_AFFECT_CURRENT
+     * 1 -> VIR_DOMAIN_AFFECT_LIVE
+     * 2 -> VIR_DOMAIN_AFFECT_CONFIG
+     */
+    let flags = Enum.VIR_DOMAIN_AFFECT_CURRENT;
+    flags |= Enum.VIR_DOMAIN_AFFECT_CONFIG;
+
+    if (newMacAddress && newMacAddress !== macAddress) {
+        console.log("Thus Happend");
+        return detachIface(macAddress, connectionName, objPath, hotplug, persistent, dispatch)
+                .then(() => {
+                    console.log("This Happend");
+                    console.log({ connectionName, vmId: objPath, mac: newMacAddress, permanent: true, hotplug: false, sourceType: networkType, source: networkSource, model: networkModel });
+                    return attachIface({ connectionName, vmId: objPath, mac: newMacAddress, permanent: persistent, hotplug, sourceType: networkType, source: networkSource, model: networkModel });
```

Comment:
Could you make the line collapse, it's too long, in my desktop screen it get's hidden :/